src/vector/_methods.py

```diff
@@ -2533,6 +2534,18 @@ def _from_signature(
 ]
 
 
+def _get_handler_index(obj: VectorProtocol) -> int:
+    """Returns the index of the first valid handler checking the list of parent classes"""
+    index = -1
+    for cls in type(obj).__mro__:
+        with suppress(ValueError):
+            index = _handler_priority.index(cls.__module__)
+            break
+
+    assert index != -1
+    return index
```

Comment:
```suggestion
    for cls in type(obj).__mro__:
        with suppress(ValueError):
            return _handler_priority.index(cls.__module__)
    raise IndexError(f"Could not find a valid handler for {obj}! This should not happen.")

    assert index != -1
    return index
```

Reply:
Shouldn't it be an AssertionError, since it would be an internal logic error, not a user error?

Otherwise, this implementation is cleaner (no need to introduce the `-1`).

Reply:
Probably, yes. It should not be an assert, as the main reason to use assert is to have the ability to turn off the assertions, but this one should never be reached, and we do always want to throw if we reach it.

Reply:
(Changed in the suggestion)

Reply:
But it's also true that an assertion represents something that should never happen. If there's ever a situation in which an assertion would fail, wrong behavior would result from letting it slip by. Turning off assertions trades safety for speed—which can be a reasonable thing to do when you're confident—but the above argument would apply to every assertion, not just this one, and then what would assertions be for?

(Note, I'm saying that it shouldn't be possible for this to get out of the `for` loop due to constraints applied elsewhere, not that it is possible and we just shouldn't allow it. That's why I think it's an assertion.)

If you're happy with it, I'll commit this suggestion.

Reply:
I am OK with it too. Feel free to commit.

Reply:
If something is never reached, turning it off doesn't speed anything up, so there's no need to turn it off. Asserts should be optional checks, not fallbacks when nothing else works. That's why the assert statement takes a condition - it's a check, not a fallback.

Unrelated to this PR, but slightly to the general discussion, Python 3.11 will finally make a `try` block virtually no-cost if there is no throw. Currently it's less expensive than an `if` if there is no throw, but non-zero.

Reply:
Actually, this is not the best way to do this, as handling the exception is expensive.

Reply:
There wouldn't be a speed difference here, since (in principle) the `raise AssertionError` line is never reached. I was referring to not having to compute expressions "`xyz`" in an `assert xyz` statement in general—the only reason I can imagine for turning off assertions.